packages/runtime/client/rest/src/contract/details.ts

```diff
@@ -43,25 +54,46 @@ export const Payout = t.type({
  * @interface
  * @category Endpoint : Get Contract By Id
  */
-export interface ContractDetails extends t.TypeOf<typeof ContractDetails> {}
+export type ContractDetails = {
+  version: MarloweVersion;
+  contractId: ContractId;
+  initialContract: Contract;
+  roleTokenMintingPolicyId: PolicyId;
+  status: TxStatus;
+  tags: Tags;
+  metadata: Metadata;
+  unclaimedPayouts: Payout[];
+} /** Only available when the contract is Unsigned/Submitted and Active */ & {
```

Comment:
@jhbertra @hrajchert , I'd like to draw your attention of these optional fields.... As a user of this endpoint, how do I know when they are available and when they are not ? It's difficult to be actionable on this data imo

Reply:
I have to go deep int the `marlowe-cardano` codebase to understand how to control these fields : 
```haskell
instance ToDTO SomeContractState where
  toDTO (SomeContractState MarloweV1 ContractState{..}) =
    Web.ContractState
      { contractId = toDTO contractId
      , roleTokenMintingPolicyId = toDTO roleTokenMintingPolicyId
      , version = Web.V1
      , tags = foldMap fst $ toDTO $ marloweMetadata metadata
      , continuations = snd =<< toDTO (marloweMetadata metadata)
      , metadata = toDTO $ transactionMetadata metadata
      , status = Web.Confirmed
      , block = Just $ toDTO initialBlock
      , initialContract = Sem.marloweContract $ datum initialOutput
      , currentContract = Sem.marloweContract . datum <$> latestOutput
      , state = Sem.marloweState . datum <$> latestOutput
      , assets = maybe emptyAssets (\Core.TransactionScriptOutput{..} -> toDTO assets) latestOutput
      , utxo = toDTO . utxo <$> latestOutput
      , txBody = Nothing
      , unclaimedPayouts =
          (\(payoutId, Core.Api.Payout{..}) -> Web.Payout (toDTO payoutId) (toDTO . tokenName $ datum) (toDTO assets))
            <$> M.toList unclaimedPayouts
      }

data ContractState v = ContractState
  { contractId :: ContractId
  , roleTokenMintingPolicyId :: PolicyId
  , metadata :: MarloweTransactionMetadata
  , initialBlock :: BlockHeader
  , initialOutput :: TransactionScriptOutput v
  , latestBlock :: BlockHeader
  , latestOutput :: Maybe (TransactionScriptOutput v)
  , unclaimedPayouts :: Map TxOutRef (Payout v)
  }
  deriving (Generic)
```

Reply:
I'm proposing something like this : 
```haskell
data ContractState = Built
  { contractId :: TxOutRef
  , roleTokenMintingPolicyId :: PolicyId
  , version :: MarloweVersion
  , tags :: Map Text Metadata
  , metadata :: Map Word64 Metadata
  , initialContract :: Semantics.Contract
  , currentContract :: Semantics.Contract
  , state :: Semantics.State
  , assets :: Assets
  , txBody :: TextEnvelope
  }
| Submitted
  { contractId :: TxOutRef
  , roleTokenMintingPolicyId :: PolicyId
  , version :: MarloweVersion
  , tags :: Map Text Metadata
  , metadata :: Map Word64 Metadata
  , block :: Maybe BlockHeader
  , initialContract :: Semantics.Contract
  , currentContract :: Semantics.Contract
  , state :: Semantics.State
  , txBody :: Maybe TextEnvelope
  }
| Active
  { contractId :: TxOutRef
  , roleTokenMintingPolicyId :: PolicyId
  , version :: MarloweVersion
  , continuations :: Maybe Text
  , tags :: Map Text Metadata
  , metadata :: Map Word64 Metadata
  , block :: BlockHeader
  , initialContract :: Semantics.Contract
  , currentContract :: Semantics.Contract
  , state :: Semantics.State
  , utxo :: TxOutRef
  , assets :: Assets
  , unclaimedPayouts :: [Payout]
  }
| Closed
  { contractId :: TxOutRef
  , roleTokenMintingPolicyId :: PolicyId
  , version :: MarloweVersion
  , tags :: Map Text Metadata
  , metadata :: Map Word64 Metadata
  , status :: TxStatus
  , block :: BlockHeader
  , initialContract :: Semantics.Contract
  , unclaimedPayouts :: [Payout]
  }
  deriving (Show, Eq, Generic)
```

Reply:
Ideally we should have `initialState` in all cases and `currentState` only in the case of `Active`.
I wonder if the state should also be reported in the `Closed` (so we can inspect the choices which were made during the execution?).